### DISCUSSION TITLE
From time to time clients get Internal Server Error

### DISCUSSION THREAD

**Discussion by MaxBazarov** [OpenIDC/mod_auth_openidc#527] [Q&A] [answered]
![0e90c9c3-8b7c-4fe3-b10f-a7229ae0c476](https://user-images.githubusercontent.com/18418306/106041668-276e7c80-60ed-11eb-853a-68de6edcadf4.jpg)
![393b3870-82a5-469a-afe2-e8e6bc5fa045](https://user-images.githubusercontent.com/18418306/106041676-29384000-60ed-11eb-9104-dc04220f1f4d.jpg)

Here is my auth_openidc.conf:

OIDCResponseType "id_token token"
OIDCScope openid
OIDCResponseMode form_post
OIDCProviderMetadataURL https://login.microsoftonline.com/<CUT>/v2.0/.well-known/openid-configuration
OIDCProviderJwksUri https://login.microsoftonline.com/<CUT>/discovery/v2.0/keys
OIDCClientID <CUT>
OIDCClientSecret <CUT>
OIDCRedirectURI https://uxteam.int.zone/oidc-redirect
OIDCCryptoPassphrase <CUT>
OIDCStateMaxNumberOfCookies 7 true
#OIDCStateTimeout 360000000
OIDCSessionInactivityTimeout 31536000
OIDCSessionMaxDuration 0
OIDCCacheType file
OIDCSessionType server-cache:persistent
Redirect permanent https://uxteam.int.zone/secure  https://uxteam.int.zone
<Location />
   AuthType openid-connect
   Require valid-user
   Require ip 157.43.177.63 47.11.145.55
<Location /_tools>
        AuthType None
        Require all granted
</Location>

How can I resolve it ? Thanks!

**Comment by zandbelt** [comment, marked as answer]
see this: https://github.com/zmartzone/mod_auth_openidc/wiki/Cookies#state-cookies-are-piling-up

**Reply by MaxBazarov** [reply]
But I don't use OIDCSessionType client-cookie !

**Reply by zandbelt** [reply]
that does not have anything to do with it, please read the page: it is about state cookies

**Reply by MaxBazarov** [reply]
Thanks.

**Comment by MaxBazarov** [comment]
After last update my config looks like this: 

OIDCStateMaxNumberOfCookies 2 true
OIDCStateTimeout 180
OIDCUnAuthAction 401 "%{HTTP_ACCEPT} =~ m#image/png#"
OIDCUnAuthAction 401 "%{HTTP_ACCEPT} =~ m#image/jpg#"
OIDCUnAuthAction 401 "%{HTTP_ACCEPT} =~ m#text/css#"
OIDCUnAuthAction 401 "%{HTTP_ACCEPT} =~ m#text/javascript#"

We don't start any Ajax code on page loading, but anyway, I added "X-Requested-With" to all other Ajax calls.

It worked fine on my tests. But today my user complains again about Bad Request.
On the attached screenshot we see many states cookies. It makes me maid.

![MicrosoftTeams-image](https://user-images.githubusercontent.com/18418306/106639320-e7e4dc00-6595-11eb-98db-efa518b02c6c.png)

**Reply by zandbelt** [reply]
FWIW: in the server side debug log you'll see what URLs those state cookies are served from so you can analyse it better than in the browser.

You'll need to use a single `OIDCUnAuthAction` primitive with a combined expression as in your current setup only the last one will prevail and actually override the default `X-Requested-With` detection algorithm...

Moreover, if all your XML HTTP request are really using `X-Requested-With` by now then you can just get rid of all `OIDCUnAuthAction` primitives as they only override the default setting that would work fine on its own.

**Reply by MaxBazarov** [reply]
Ok, thanks for OIDCUnAuthAction information. I will apply changes.
But the question is still the same. Why user does get soo many state cookies?
